compiler/rustc_resolve/src/late.rs

```diff
@@ -3515,6 +3538,18 @@ impl<'a: 'ast, 'b, 'ast> LateResolutionVisitor<'a, 'b, 'ast> {
                 });
             }
             // For closures, ClosureOrAsyncRibKind is added in visit_fn
+            ExprKind::Closure(ClosureBinder::For { id, ref generic_params, span }, ..) => {
+                self.with_generic_param_rib(
+                    &generic_params,
+                    NormalRibKind,
+                    LifetimeRibKind::Generics {
+                        binder: id,
+                        kind: LifetimeBinderKind::Function,
```

Comment:
Could you create a dedicated variant?

Reply:
Done in e13019db5c05b0d0b96e7c3444eaaec64a582870